src/Apps/Order/Routes/Shipping2/Components/SavedAddresses2.tsx

```diff
@@ -1,148 +1,177 @@
   const logger = createLogger("SavedAddresses.tsx")
-  const [activeModal, setActiveModal] = useState<AddressModalAction | null>(
-    null
-  )
+
   const shippingContext = useShippingContext()
+  const orderTracking = useOrderTracking()
+  const formikContext = useFormikContext<FulfillmentValues>()
 
-  const { onSelect, me, relay } = props
+  const [
+    addressModalAction,
+    setAddressModalAction,
+  ] = useState<AddressModalAction | null>(null)
 
-  const addressList = compact<SavedAddressType>(
-    extractNodes(me?.addressConnection) ?? []
+  const data = useFragment(
+    graphql`
+      fragment SavedAddresses2_me on Me
+        @argumentDefinitions(
+          first: { type: "Int", defaultValue: 30 }
+          last: { type: "Int" }
+          after: { type: "String" }
+          before: { type: "String" }
+        ) {
+        addressConnection(
+          first: $first
+          last: $last
+          before: $before
+          after: $after
+        ) {
+          edges {
+            node {
+              internalID
+              name
+              addressLine1
+              addressLine2
+              city
+              region
+              postalCode
+              country
+              phoneNumber
+              isDefault
+            }
+          }
+        }
+      }
+    `,
+    props.me
   )
 
-  const selectedSavedAddressId =
-    shippingContext.orderData.savedFulfillmentDetails?.selectedSavedAddressId
-  const [selectedAddressID, setSelectedAddressID] = useState<
-    string | undefined
-  >(
-    getBestAvailableAddress(
-      addressList,
-      selectedSavedAddressId,
-      shippingContext.orderData.availableShippingCountries
-    )?.internalID
+  const addressList = compact<SavedAddressType>(
+    extractNodes(data?.addressConnection) ?? []
   )
 
-  const selectedAddress =
-    selectedAddressID && getAddressByID(addressList, selectedAddressID)
-  const selectedAddressPresent = !!selectedAddress
-
-  const orderTracking = useOrderTracking()
+  const savedAddressOnOrder = shippingContext.orderData.savedFulfillmentDetails
+    ?.selectedSavedAddressID
+    ? getAddressByID(
+        addressList,
+        shippingContext.orderData.savedFulfillmentDetails
+          ?.selectedSavedAddressID
+      ) ?? null
+    : null
+
+  const selectAndSubmitAddress = (address: SavedAddressType) => {
+    shippingContext.actions.setSelectedSavedAddressID(address.internalID)
+    props.onSelect(address)
+  }
 
-  React.useEffect(() => {
-    if (!selectedAddressPresent) {
-      setSelectedAddressID(
-        getBestAvailableAddress(
+  const addressSavedToOrderID = savedAddressOnOrder?.internalID
+
+  // TODO: Make sure this can't create an infinite loop if submitting fails
+  // Is it necessary beyond the initial render?
+  useEffect(() => {
+    // Automatically select (save) best available address ID if it isn't present
+    const automaticallySelectBestAddress = async () => {
```

Comment:
To avoid endless submitting if something goes wrong (eg the saved address is not shippable for this artwork) i only put `[props.active]` in the dependency array.